impl/src/main/java/dao/QuestionDao.java

```diff
@@ -49,13 +52,91 @@
             ":limit")
     Observable<Question> getLatestQuestions(Integer limit);
 
+    @Query(
+        "SELECT " +
+            "question.id, " +
+            "question.question, " +
+            "answer_accepted, " +
+            "question.title, " +
+            "question.bounty, " +
+            "question.created_at, " +
+            "question.user_id, " +
+            "question.slack_id, " +
+            "\"user\".picture, " +
+            "\"user\".name as created_by, " +
+            "(SELECT COALESCE(SUM(question_vote.value), 0) FROM question_vote WHERE question_vote.question_id = question.id) AS votes " +
+        "FROM " +
+            "question " +
+        "INNER JOIN " +
+            "\"user\" on \"user\".id = question.user_id " +
+        "ORDER BY " +
+            "votes DESC NULLS LAST, " +
+            "question.created_at DESC " +
+        "LIMIT " +
+            ":limit")
+    Observable<Question> getPopularQuestions(Integer limit);
+
+    @Query(
+        "SELECT " +
+            "question.id, " +
```

Comment:
Man kanske kan bryta ut lite konstanter här som man kan återanvända.

Reply:
I tried that at one point, but @splitfeed had strong opinions against. 😆

Reply:
I agree that the query annotation should contain the complete SQL expression. Concatenating stuff is just confusing and prone to errors (when changing the value of a constant).